libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -2749,6 +2749,10 @@ std::string IRGeneratorForStatements::shiftOperation(
 	IRVariable const& _amountToShift
 )
 {
+	solUnimplementedAssert(
+		_amountToShift.type().category() != Type::Category::FixedPoint,
```

Comment:
```suggestion
		_amountToShift.type().category() != Type::Category::FixedPoint &&
		_value.type().category != Type::Category::FixedPoint
```
Probably saves another fixed point PR :)

Reply:
These two are actually added in the yul util function, so I think it is fine for them at least

Reply:
Ah, I see.

Reply:
Oops, looks like @hrkrshnn was right:

```
contract C {
        function f() public pure {
                fixed x;
                assert(x >>> 6 == 0);
        }
}
```

passes this assertion but would have failed `_value.type().category != Type::Category::FixedPoint`

Not sure why this percolated through to the generator.

Reply:
Oh no!

Reply:
But it would hide actual failure behind unimplemented fixed point types. 
It fails here `solAssert(_operator == Token::SHL || _operator == Token::SAR, "");`

Reply:
But, will add like @hrkrshnn suggested at the end. Once we implement FixedPoint shifts, we can look at the other one...

Reply:
Actually how about we directly give an unimplemented error inside, say type checker whenever any `fixed` type is seen. This would at least prevent all these fuzzing problems. (Maybe further up the stack, even during parsing.) What do you think @mijovic @bshastry

Reply:
Sounds good to me. I could catch unimplemented assertions in the fuzz API so they are not even reported as crashes.

Reply:
Ok. Then I'll make a PR for that.